src/Microsoft.AspNet.OData/EnableQueryAttribute.cs

```diff
@@ -27,6 +27,7 @@ namespace Microsoft.AspNet.OData
     /// <see cref="EnableQueryAttribute"/> to validate incoming queries. For more information, visit
     /// http://go.microsoft.com/fwlink/?LinkId=279712.
     /// </summary>
+    [SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling", Justification = "The majority of types referenced by this method result from HttpActionExecutedContext")]
```

Comment:
Is it really necessary to do this at class level? Or can it only be applied to the specific method(s) that are failing?

Reply:
@odero, I think the metric this rules check can be calculated at method and class level. Is failing at method level (in the ones that has the supress warning), but also is failing at class level (so even if we put the supress warning in other methods, fails).

Reply:
@odero, regarding the test, we don't have an automated test to show this issue, just a sample project attached to the issue. I'm leaving office today and won't be able to produce a ticket until 2 February.

Reply:
+1

---
In reply to: [368624365](https://github.com/OData/WebApi/pull/2014#discussion_r368624365) [](ancestors = 368624365)